src/test/util/matchable.rs

```diff
@@ -67,8 +67,7 @@ impl Matchable for Document {
             if k == "afterClusterTime" {
                 continue;
             }
-            // TODO RUST-97: Remove this logic to bypass recoveryToken
-            if k == "recoveryToken" {
+            if k == "recoveryToken" && v.is_placeholder() && self.get_document(k).is_ok() {
```

Comment:
I think this isn't needed since `.matches` accounts for placeholder already.

Reply:
`.matches` checks that the placeholder exists, but not that the underlying value is a BSON Document, which the spec requires.